test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/ReadOnlyTagHelperAttributesTest.cs

```diff
@@ -0,0 +1,643 @@
+                    { new[] { A, B, C }, "cnamE", new[] { C }, true },
+
+                    // Multiple elements same name
+                    { new[] { A, B, A2, C }, "AName", new[] { A, A2 }, true },
+                    { new[] { A, B, A2, C }, "aname", new[] { A, A2 }, true },
+                    { new[] { B, A2, A }, "aname", new[] { A2, A }, true },
+                    { new[] { B, A2, A, C }, "AName", new[] { A2, A }, true },
+                    { new[] { A, A3 }, "AName", new[] { A, A3 }, true },
+                    { new[] { A3, A }, "aname", new[] { A3, A }, true },
+                    { new[] { A, A2, A3 }, "AName", new[] { A, A2, A3 }, true },
+                    { new[] { C, B, A3, A }, "AName", new[] { A3, A }, true },
+
+                    // Null expected lookups
+                    { new[] { A }, "_AName_", Enumerable.Empty<IReadOnlyTagHelperAttribute>(), false },
+                    { new[] { A, B }, "_AName_", Enumerable.Empty<IReadOnlyTagHelperAttribute>(), false },
+                    { new[] { A, B, C }, "_BName_", Enumerable.Empty<IReadOnlyTagHelperAttribute>(), false },
+                    { new[] { A, A2, B, C }, "_cnamE_", Enumerable.Empty<IReadOnlyTagHelperAttribute>(), false },
+                    { new[] { A, A2, A3, B, C }, "_cnamE_", Enumerable.Empty<IReadOnlyTagHelperAttribute>(), false },
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(TryGetAttributesData))]
+        public void TryGetAttributes_ReturnsExpectedValueAndAttribute(
+            IEnumerable<IReadOnlyTagHelperAttribute> initialAttributes,
+            string nameToLookup,
+            IEnumerable<IReadOnlyTagHelperAttribute> expectedAttributes,
+            bool expectedResult)
+        {
+            // Arrange
+            var attributes = new ReadOnlyTagHelperAttributes<IReadOnlyTagHelperAttribute>(initialAttributes);
+            IEnumerable<IReadOnlyTagHelperAttribute> resolvedAttributes;
+
+            // Act
+            var result = attributes.TryGetAttributes(nameToLookup, out resolvedAttributes);
+
+            // Assert
+            Assert.Equal(expectedResult, result);
+            Assert.Equal(expectedAttributes, resolvedAttributes, CaseSensitiveTagHelperAttributeComparer.Default);
+        }
+
+        [Fact]
+        public void Attributes_EqualsInitialAttributes()
+        {
+            // Arrange
+            var expectedAttributes = new[]
+            {
+                new TagHelperAttribute("A", "AV"),
+                new TagHelperAttribute("B", "BV")
+            };
+
+            // Act
+            var attributes = new CustomAttributes(expectedAttributes);
+
+            // Assert
+            Assert.Equal(expectedAttributes, attributes.ExposedAttributes);
+        }
+
+        [Fact]
+        public void GetEnumerator_ReturnsUnderlyingAttributesEnumerator()
+        {
+            // Arrange & Act
+            var attributes = new CustomAttributes(new[]
```

Comment:
no need to use the `Testable` class here.  just use `new ReadOnlyTagHelperAttributes<IReadOnlyTagHelperAttribute>(...)`